src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -1140,12 +1140,7 @@ public final ForkJoinWorkerThread newThread(ForkJoinPool pool) {
             boolean isCommon = (pool.workerNamePrefix == null);
             @SuppressWarnings("removal")
             SecurityManager sm = System.getSecurityManager();
-            if (sm == null) {
-                if (isCommon)
-                    return new ForkJoinWorkerThread.InnocuousForkJoinWorkerThread(pool);
-                else
-                    return new ForkJoinWorkerThread(null, pool, true, false);
-            } else if (isCommon)
+            if (sm != null && isCommon)
```

Comment:
For common thread pool, if SM is not enabled, it will create `ForkJoinWorkerThread` that does not clear thread locals which is different than JDK 18 behavior.  Is this intentional?

Reply:
@mlchung Yeah, @DougLea perhaps want to weigh in on that. I wonder if this should be addressed as this PR or if it should be handled separately?

Reply:
@mlchung I think we should proceed with merging this as-is and look to address the ThreadLocal situation (or not at all, based on @DougLea's input). Are you OK with reviewing this as-is?

Reply:
Yes it's fine to separate it as a follow up.